ste/mgr-JobPartTransferMgr.go

```diff
@@ -158,6 +161,28 @@ func (jptm *jobPartTransferMgr) GetOverwriteOption() common.OverwriteOption {
 	return jptm.jobPartMgr.GetOverwriteOption()
 }
 
+func (jptm *jobPartTransferMgr) ShouldDecompress() bool {
+	if jptm.jobPartMgr.AutoDecompress() {
+		ct, _ := jptm.GetSourceCompressionType()
+		return ct != common.ECompressionType.None()
+	}
+	return false
+}
+
+func (jptm *jobPartTransferMgr) GetSourceCompressionType() (common.CompressionType, error) {
+	encoding := jptm.Info().SrcHTTPHeaders.ContentEncoding
+	switch strings.ToLower(encoding) {
+	case "":
+		return common.ECompressionType.None(), nil
+	case "gzip":
+		return common.ECompressionType.GZip(), nil
+	case "deflate":
+		return common.ECompressionType.ZLib(), nil
+	default:
```

Comment:
Consider just logging that it's unsupported instead of erroring out.

Reply:
Once were in the STE (rather than enumerators), erroring out is actually the right thing to do - because it only errors us out of that one transfer. I.e. it will fail the transfer, but not the job. Does that make sense, and address what you were thinking of in your comment?

Reply:
That makes _sense_, but what I'm saying is more along the lines of "let the transfer continue, but warn the user that there's no special decompression happening"

Reply:
I don't think that's the right thing to do.  In general, we should do exactly as the user asks, or else fail in a _small number_ of _easily identified_ ways(*).  Generally, textual warnings to the user don't meet those criteria - since on a large job with 100's of thousands of files, its not practical for a user to read and respond to a large number of warnings on screen.  And it's also not practical for them to read and respond to a large number of warnings in the log, since what would they search for, and how would they know that they even need to do such a search?

(*) I know we already have to break this "rule" at various places in the enumeration code.  But so far we are not routinely breaking it in the STE, and I don't want to start breaking it in the STE.

I'm gonna close this thread with no change to the code, since that's the only option that will leave it consistent with policy used in the rest of the STE.